index.js

```diff
@@ -13,23 +13,55 @@ const isAvailable = options => new Promise((resolve, reject) => {
 	});
 });
 
-const getPort = options => {
-	options = Object.assign({}, options);
+const portCheckSequence = function * (ports) {
+	if (ports) {
+		yield * ports;
+	}
+
+	yield 0; // Fall back to 0 if anything else failed
+};
+
+module.exports = async options => {
+	let ports = null;
 
-	if (typeof options.port === 'number') {
-		options.port = [options.port];
+	if (options) {
+		ports = (typeof options.port === 'number') ? [options.port] : options.port;
 	}
 
-	return (options.port || []).reduce(
-		(seq, port) => seq.catch(
-			() => isAvailable(Object.assign({}, options, {port}))
-		),
-		Promise.reject()
-	);
+	for (const port of portCheckSequence(ports)) {
+		const gotPort = await getAvailablePort({...options, port}); // eslint-disable-line no-await-in-loop
```

Comment:
If we end up reverting the change introduced in `getAvailablePort`:

```
try {
   return await getAvailablePort({...options, port});
} catch (err) {
   // safely swallow err to continue the loop  
}
```

Reply:
I wonder why you find this normal, but ```js
server.on('error', err => (err.code === 'EADDRINUSE') ? resolve(null) : reject(err));
``` counter-intuitive. It's the same piece of code(which will probably be even compiled to exactly same sequence except debug symbols).

Reply:
But anyhow, now that `getAvailablePort` is being called only in one place, it can be done.